Assets/_Laz/Scripts/UserInterface/DebugBehaviour/DebugLazMovement/DebugMovementParametersEditor.cs

```diff
@@ -0,0 +1,312 @@
+using UnityEngine;
+
+namespace Laz
+{
+    [RequireComponent(typeof(RectTransform))]
+    public class DebugMovementParametersEditor : MonoBehaviour
+    {
+        private const float HEADER_HEIGHT_MULTIPLIER = 0.1f;
+        private const float BODY_HEIGHT_MULTIPLIER = 0.7f;
+        private const float FOOTER_HEIGHT_MULTIPLIER = 0.15f;
+        private const float HEADER_PADDING_MULTIPLIER = 0.03f;
+
+        private const float MIN_CURVATURE_RATE = 0.005f;
+        private const float MAX_CURVATURE_RATE = 0.1f;
+
+        [SerializeField] private RectTransform _rectTransform;
+        private LazMovementPropertyScriptableObject _movementProperty;
+        
+        private Rect _rect;
+        private string _fileName = string.Empty;
+
+        private float _acceleration = 1f;
+        private float _deceleration = 1f;
+        private float _baseMaxSpeed = 5f;
+        private float _curvatureRate = 0.1f;
+        private float _lazMaxSpeed = 10f;
+        private float _boostTimeLimit = 2f;
+        private float _boostSpeed = 20f;
+
+        private bool _isPanelOpen = false;
+        
+        private Vector2 _scrollPosition;
+
+        //GUIStyles
+        private GUIStyle _headerGUIStyle = GUIStyle.none;
+        private GUIStyle _labelGUIStyle = GUIStyle.none;
+        private GUIStyle _valueGUIStyle = GUIStyle.none;
+        private GUIStyle _fileNameStyle = GUIStyle.none;
+        private GUIStyle _footerGUIStyle = GUIStyle.none;
+
+        //GUIParameters
+        private float _headerHeight;
+        private float _headerPadding;
+        private float _bodyHeight;
+        private float _footerHeight;
+
+        public void OpenWindow()
+        {
+            Initialize();
+        }
+
+        public void CloseWindow()
+        {
+            gameObject.SetActive(false);
+        }
+        
+        private void Awake()
+        {
+            CreateGUIParameters();
+            gameObject.SetActive(false);
+        }
+
+        private void Initialize()
+        {
+            ResetContent();
+            GetMovementProperty();
+            gameObject.SetActive(true);
+        }
+
+        private void GetMovementProperty()
+        {
+            LazCoordinatorBehaviour coordinator = FindObjectOfType<LazCoordinatorBehaviour>();
+
+            if (coordinator == null)
+            {
+                Debug.LogError("No LazCoordinatorBehaviour found in scene, unable to process Movement Property");
+                CloseWindow();
+            }
+
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            _movementProperty = coordinator.LazMovementPropertyScriptableObject;
+#endif
```

Comment:
I'm having some issues in the scene, where my changes seem very inconsistently actually applied to Laz. I'm not sure what's causing it, but do you maybe need to be applying the movementProperty back to the coordinator or something like that?